### ISSUE TITLE
Toastr doesn't dissappear if we use the latest version of React (18.1.0), the remove toast action doesn't get call

### ISSUE THREAD

**Issue by uditkhanna112** [diegoddox/react-redux-toastr#286]
If we render the component using 👍
 const root = ReactDOM.createRoot(
  document.getElementById("root") as HTMLElement
);
root.render(
  <React.StrictMode>
    <Provider store={store}>
      <ReduxToastr
        // preventDuplicates
        timeOut={4000}
        confirmOptions={{
          okText: "Yes",
          cancelText: "No",
        }}
        position="bottom-right"
        newestOnTop={true}
        progressBar
      />
      <App />
    </Provider>
  </React.StrictMode>
);

Then the toastr doesn't dissappear, using old way of reactdom.render it's working fine.

**Comment by uditkhanna112** [comment]
@diegoddox

**Comment by diegoddox** [comment]
Thanks for reporting @uditkhanna112 not sure when I will be able to look at this issue, any help is more than welcome 🙏

**Comment by uditkhanna112** [comment]
The issue persists, i had a workaround for it like using setTimeout and removing the toastrs by dispatching remove actions but again that's not really a good optimized solution, there is a big dependency of this package on our project, please look into it if you have a better solution

**Comment by uditkhanna112** [comment]
@diegoddox @Mordred @icidasset @jalkoby

**Comment by jefferdo** [comment]
@diegoddox 

I've noticed an issue in the ToastrBox component when using React 18. The toasts were not being removed as expected due to changes in how React 18 handles and batches updates, which affected the `onCSSTransitionEnd` function.

To resolve this, I've created a pull request where I replaced the `onCSSTransitionEnd` function with direct usage of the `transitionend` event listener. The event listener is added in the `componentDidMount` method and removed in the `componentWillUnmount` method. This ensures that the `_onAnimationComplete` method is called when a transition ends on `this.toastrBoxElement`, even in React 18.

I've also added checks to ensure `this.toastrBoxElement` is not `null` before adding or removing the event listener. This prevents potential errors if `this.toastrBoxElement` is `null`.

With these changes, toasts should now be removed as expected in React 18. I've created a pull request for these changes and I hope this contribution helps improve the project. Looking forward to your feedback.
